api/internal/plugins/loader/loader_no_go.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2022 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:build no_go_plugins
+// +build no_go_plugins
+
+package loader
+
+import (
+	"fmt"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/kyaml/resid"
+)
+
+func (l *Loader) loadGoPlugin(id resid.ResId, absPath string) (resmap.Configurable, error) {
+	return nil, fmt.Errorf("kustomize was built without go plugin support (-tags=no_go_plugins)")
```

Comment:
Is this used anywhere that could possibly break a normal run of `kustomize build`? I'm wondering if there is a benefit of returning this error here vs just doing nothing, i.e. `return nil, nil`.

I'm also trying to think about if it's smart to have a separate build file in the loader. If we do this, we would probably want a github action that runs our non-plugin tests, right? This gives us a whole other build version of kustomize to think about and maintain. 

@KnVerey WDYT? I'm very hesitant, but I don't know if there are other ways to approach resolving the issue.

Reply:
> Is this used anywhere that could possibly break a normal run of kustomize build?

I looked for all usages and the loader seems to just be used here. We could get more specific with the build tag to avoid overlaps. 

https://cs.k8s.io/?q=loadGoPlugin&i=nope&files=&excludeFiles=&repos=kubernetes-sigs/kustomize

> I'm wondering if there is a benefit of returning this error here vs just doing nothing, i.e. return nil, nil

I think it's worth explicitly erroring if a users tries to load a Go plugin with the support disabled. Or panicking even.

> I'm also trying to think about if it's smart to have a separate build file in the loader. If we do this, we would probably want a github action that runs our non-plugin tests with this build tag set, right? This gives us a whole other build version of kustomize to think about and maintain.

What's the timeline for deprecating Go plugins? The problem @howardjohn ran into seems to be very very niche. I wouldn't even think it's worth documenting unless we hear it from other folks.

Reply:
This feel off my radar until the bot closed it, but I agree with Natasha that we'd need test coverage for both versions of the build, and with Eddie that the maintenance burden of that (and actual CI cost) is not worth it to avoid warnings (not errors) in a very, very niche use case.